### PR TITLE
Implement functions for calling ProxyReader.GetMany from Unstoppable Domains

### PR DESCRIPTION
**Pull request by yrliou**
Resolves https://github.com/brave/brave-browser/issues/15126
Easier to review per commit.

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/handbook/blob/master/development/security.md#when-is-a-security-review-needed), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

File: components/brave_wallet/browser/eth_json_rpc_controller.cc [hunk 1]
```diff
@@ -240,6 +240,41 @@ void EthJsonRpcController::OnGetERC20TokenBalance(
...
+  if (!unstoppable_domains::GetMany(keys, domain, &data)) {
+    return false;
+  }
+
+  Request(eth_call("", contract_address, "", "", "", data, "latest"),
+          std::move(internal_callback), true);
+  return true;
+}
+
+void EthJsonRpcController::OnUnstoppableDomainsProxyReaderGetMany(
```
Comment (bbondy) [inline components/brave_wallet/browser/eth_json_rpc_controller.cc:261]: Could we add some browsertests for these in `browser/brave_wallet/eth_json_rpc_controller_browsertest.cc`?

File: components/brave_wallet/browser/brave_wallet_utils_unittest.cc [hunk 2]
```diff
@@ -273,4 +273,284 @@ TEST(BraveWalletUtilsUnitTest, Mnemonic) {
   }
 }
 
+TEST(BraveWalletUtilsUnitTest, EncodeString) {
+  std::string output;
+  EncodeString("one", &output);
```
Comment (bbondy) [inline components/brave_wallet/browser/brave_wallet_utils_unittest.cc:278]: ASSERT_TRUE here and for other `EncodeString` calls

File: components/brave_wallet/browser/brave_wallet_utils_unittest.cc [hunk 3]
```diff
@@ -273,4 +273,284 @@ TEST(BraveWalletUtilsUnitTest, Mnemonic) {
...
+      "6f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e"
+      "656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f6e656f"
+      "6e65000000000000000000000000000000000000000000000000");
+
+  output.clear();
+  EncodeString("", &output);
+  EXPECT_EQ(
+      output,
+      "0x0000000000000000000000000000000000000000000000000000000000000000");
+}
```
Comment (bbondy) [inline components/brave_wallet/browser/brave_wallet_utils_unittest.cc:304]: Would also be good to include some invalid input calls, I guess at least `!IsStringUTF8`

File: components/brave_wallet/browser/brave_wallet_utils_unittest.cc [hunk 4]
```diff
@@ -273,4 +273,284 @@ TEST(BraveWalletUtilsUnitTest, Mnemonic) {
...
+            "0000000000000000000000000000000000000000000000000000000000000003"
+            // encoding for "two"
+            "74776f0000000000000000000000000000000000000000000000000000000000"
+            // count for ""
+            "0000000000000000000000000000000000000000000000000000000000000000"
+            // count for "three"
+            "0000000000000000000000000000000000000000000000000000000000000005"
+            // encoding for "three"
+            "7468726565000000000000000000000000000000000000000000000000000000");
+}
```
Comment (bbondy) [inline components/brave_wallet/browser/brave_wallet_utils_unittest.cc:398]: Ditto a test where `EncodeStringArray` returns false

File: components/brave_wallet/browser/brave_wallet_utils_unittest.cc [hunk 5]
```diff
@@ -273,4 +273,284 @@ TEST(BraveWalletUtilsUnitTest, Mnemonic) {
...
+  EXPECT_EQ(output, "one two three four five six seven eight nine");
+
+  output.clear();
+  EXPECT_TRUE(DecodeString(
+      0,
+      // count for ""
+      "0000000000000000000000000000000000000000000000000000000000000000",
+      &output));
+  EXPECT_EQ(output, "");
+}
```
Comment (bbondy) [inline components/brave_wallet/browser/brave_wallet_utils_unittest.cc:429]: Ditto invalid `DecodeString` result

File: components/brave_wallet/browser/brave_wallet_utils_unittest.cc [hunk 6]
```diff
@@ -273,4 +273,284 @@ TEST(BraveWalletUtilsUnitTest, Mnemonic) {
...
+      // count for ""
+      "0000000000000000000000000000000000000000000000000000000000000000"
+      // count for "three"
+      "0000000000000000000000000000000000000000000000000000000000000005"
+      // encoding for "three"
+      "7468726565000000000000000000000000000000000000000000000000000000",
+      &output));
+  expected_output = {"", "one", "", "two", "", "three"};
+  EXPECT_EQ(output, expected_output);
+}
```
Comment (bbondy) [inline components/brave_wallet/browser/brave_wallet_utils_unittest.cc:524]: Ditto `DecodeStringArray` results

**Review by bbondy — COMMENTED** [review]
Looks good, just some minor test related requests.

**Comment by bbondy** [comment]
nice work

File: components/brave_wallet/browser/eth_json_rpc_controller.cc (hunk 1 above)
Comment (yrliou) [reply components/brave_wallet/browser/eth_json_rpc_controller.cc:261]: Added in https://github.com/brave/brave-core/pull/8437/commits/732166ca4493d18df5685812f7ade76210d8f71d

File: components/brave_wallet/browser/brave_wallet_utils_unittest.cc (hunk 2 above)
Comment (yrliou) [reply components/brave_wallet/browser/brave_wallet_utils_unittest.cc]: This and below comments are addressed in https://github.com/brave/brave-core/pull/8437/commits/2ff5dfe688cf52584c76ed97398246637ea2d0fc.

**Comment by yrliou** [comment]
@bbondy Ready to review again, only last two commits, https://github.com/brave/brave-core/pull/8437/commits/2ff5dfe688cf52584c76ed97398246637ea2d0fc and https://github.com/brave/brave-core/pull/8437/commits/732166ca4493d18df5685812f7ade76210d8f71d, are new after you last reviewed, thanks.

**Review by bbondy — APPROVED** [review]

**Comment by yrliou** [comment]
linux unrelated browser test failure: 
```
[2021-04-07T22:44:26.381Z] 1 test crashed:

[2021-04-07T22:44:26.381Z]     RewardsPublisherBrowserTest.VisitVerifiedPublisher (../../brave/components/brave_rewards/browser/test/rewards_publisher_browsertest.cc:138)
```

**Pull request merged**